compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -181,6 +181,7 @@ pub struct Parser<'a> {
     /// The previous token.
     pub prev_token: Token,
     pub capture_cfg: bool,
+    pub parse_cfg_pred: bool,
```

Comment:
I don't think this is a good approach, it turns a local problem (`interface::parse_cfg` swallowing legitimate errors by using fatal-only emitter) into a concern for the whole parsing infrastructure.

What prevents `parse_cfg` from using a regular non-fatal-only emitter?
Or at least setting `Recovery::Forbidden` in the `Parser`?

Reply:
`interface::parse_cfg` won't use it in fact, becasue it will ignore errors during parsing, and the new `interface::parse_cfg`  still just check the final recovered result and then emit proper errors.

And this flag works when we parsing `#[cfg(...)]` and then call `parse_meta_attr`, `parse_cfg_pred` will make the first is parsed by `parse_ident` instead of `parse_path`.

Reply:
> I'd rather make the diagnostics worse again than preserve the bugs.

I will try to use a regular emitter and check if there will be any problems.

Reply:
> so using a post-processing check after parsing is fine, I think it's better to keep doing that.

Current impl is because I tried to support raw path-segement kw. And the post-precessing check cannot know whether the source is raw or now, so it will deny all usage although raw.

Then I use current way to deny non-raw them when parsing. But I find raw path-segement kw will destroy many things.

Soooo, I agree and will recover the old impl.